packages/yoastseo/src/languageProcessing/languages/pt/helpers/internal/getParticiples.js

```diff
@@ -1,8 +1,33 @@
-import { includes } from "lodash-es";
 import { languageProcessing } from "yoastseo";
 const { getWords } = languageProcessing;
 
-import participles from "../../config/internal/passiveVoiceParticiples";
+import participleStems from "../../config/internal/passiveVoiceParticiples";
+
+/**
+ * Checks if a Portuguese word is a participle.
+ * @param {string} word A Portuguese word.
+ * @returns {boolean} Returns true if word is a Portuguese participle, and false otherwise.
+ */
+function isParticiple( word ) {
+	// Check if the word ends with one of the vowel endings
+	if ( word.endsWith( "a" ) || word.endsWith( "o" ) ) {
+		// Remove the suffix from the word
+		const stem = word.slice( 0, -1 );
+		// Check if the stem is in the list of participles
+		if ( participleStems.includes( stem ) ) {
+			return true;
+		}
+		// Check if the word ends with one of the vowel endings
+	} else if ( word.endsWith( "as" ) || word.endsWith( "os" ) ) {
+		// Remove the suffix ending from the word
+		const stem = word.slice( 0, -2 );
+		// Check if the stem is in the list of participles
+		if ( participleStems.includes( stem ) ) {
+			return true;
+		}
+	}
+	return false;
```

Comment:
This is a good approach. But I think it can be optimized even further, since we're actually running the exact same check in both `if` blocks. So something like:
```
	const participleSuffixes = [ "a", "o", "as", "os" ];
	// For each participle suffixes, check if the word ends in one of them.
	return participleSuffixes.some( suffix => {
	   if ( word.endsWith( suffix ) ) {
	      // If the word ends with one of the suffixes, retrieve the stem.
	        const stem = word.slice( 0, -( suffix.length ) );
		// Check if the stem is in the list of participles: return true if it is, otherwise return false.
		return participleStems.includes( stem );
           }
        } );
```

This way, we don't need to repeat the logic for checking whether the stem is in the list of participle stems :)